modules/entity_version_history/tests/FunctionalJavascript/HistoryTabSettingsTest.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace Drupal\Tests\entity_version_history\Functional;
+
+use Drupal\FunctionalJavascriptTests\WebDriverTestBase;
+use Drupal\Tests\SchemaCheckTestTrait;
+
+/**
+ * Ensures the entity version history config is correctly saved.
+ *
+ * @group entity_version_history
+ */
+class HistoryTabSettingsTest extends WebDriverTestBase {
+
+  use SchemaCheckTestTrait;
+
+  /**
+   * Modules to enable.
+   *
+   * @var array
+   */
+  public static $modules = [
+    'node',
+    'entity_version',
+    'entity_version_workflows',
+    'entity_version_workflows_example',
+    'entity_version_history',
+    'menu_link_content',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected $defaultTheme = 'stark';
+
+  /**
+   * A user with administrative permissions.
+   *
+   * @var \Drupal\user\UserInterface
+   */
+  protected $adminUser;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    // Create user.
+    $this->adminUser = $this->drupalCreateUser([
+      'access entity version history configuration',
+      'access administration pages',
+    ]);
+    $this->drupalLogin($this->adminUser);
+  }
+
+  /**
+   * Tests whether the history settings form is correctly saving the settings.
+   */
+  public function testHistoryTabSettingsForm() {
+    $this->drupalGet('admin/config/entity-version/history-tab');
+    $page = $this->getSession()->getPage();
+    $assert_session = $this->assertSession();
+
+    $entity_checkbox = $assert_session->elementExists('css', '#edit-entity-types-node');
+    $entity_checkbox->check();
+    $bundle_checkbox = $assert_session->waitForElementVisible('css', '#edit-node-entity-version-workflows-example');
```

Comment:
Do we need to actually wait for this? have you encountered problems? States works pretty fast.